api/src/main/java/org/pmiops/workbench/firecloud/FirecloudApiClientFactory.java

```diff
@@ -97,4 +97,41 @@ public ApiClient newApiClient() {
         .setReadTimeout(workbenchConfig.firecloud.timeoutInSeconds, TimeUnit.SECONDS);
     return apiClient;
   }
+
+  /**
+   * Creates a Rawls API client authenticated as the given user via impersonation. This should be
+   * used as a last resort, and only when the following conditions hold:
+   *
+   * <ol>
+   *   <li>The API client is used outside the context of an end user request (exception: admin
+   *       requests)
+   *   <li>RW service account credentials lack sufficient permission to make the desired call
+   * </ol>
+   *
+   * Note that impersonated API calls are subject to the same constraints as end user calls,
+   * including access tier membership.
+   */
+  public org.pmiops.workbench.rawls.ApiClient newImpersonatedRawlsApiClient(String userEmail)
+      throws IOException {
+    OAuth2Credentials delegatedCreds = getDelegatedUserCredentials(userEmail);
+    org.pmiops.workbench.rawls.ApiClient client = newRawlsApiClient();
+    client.setAccessToken(delegatedCreds.getAccessToken().getTokenValue());
+    return client;
+  }
+
+  /**
+   * Creates a Rawls API client, unauthenticated. Most clients should use an authenticated, request
+   * scoped bean instead of calling this directly.
+   */
+  public org.pmiops.workbench.rawls.ApiClient newRawlsApiClient() {
+    WorkbenchConfig workbenchConfig = workbenchConfigProvider.get();
+    org.pmiops.workbench.rawls.ApiClient apiClient = new org.pmiops.workbench.rawls.ApiClient();
+    apiClient.setBasePath(workbenchConfig.firecloud.baseUrl);
+    apiClient.addDefaultHeader(X_APP_ID_HEADER, workbenchConfig.firecloud.xAppIdValue);
+    apiClient.setDebugging(workbenchConfig.firecloud.debugEndpoints);
+    apiClient
+        .getHttpClient()
+        .setReadTimeout(workbenchConfig.firecloud.timeoutInSeconds, TimeUnit.SECONDS);
+    return apiClient;
+  }
```

Comment:
done. I did that before, then revert back because I was thinking to put all firecloud api clients (not just firecloud-orch) there because it is simpler to manage. But it is getting confused as we have firecloud = firecloud-orch but actually firecloud should include all Terra services...
Consolidate all Terra calls might be a good next step.

Reply:
That makes sense. We can revisit this once we move off firecloud-orch.